File: src/commands/server_side/ShowCommand.java
package commands.server_side;

import OrganizationDatabase.OrganizationManagerInterface;
import lib.ExecutionStatus;
import network.client.udp.User;

public class ShowCommand extends ServerSideCommand {

    public ShowCommand(ServerCallbackFunction callback) {
        super(callback);
    }

    @Override
    protected void executeImplementation(
            User user,
            OrganizationManagerInterface organizationManager,
            Object[] args,
            ServerCallbackFunction callback
    ) throws Exception {
        assert args.length == 0;

        // TODO: add argument to the show command
        callback.invoke(
                user,
                organizationManager,
                ExecutionStatus.SUCCESS,
                null,
                organizationManager.toYaml()
        );
    }
}

File: src/commands/server_side/ShowCommand.kt
package commands.server_side

import database.OrganizationManagerInterface
import exceptions.InvalidOutputFormatException
import network.client.udp.User

class ShowCommand : ServerSideCommand() {
    override suspend fun executeImplementation(
        user: User?,
        organizationManager: OrganizationManagerInterface,
        argument: Any?
    ): Result<String> {
        return when (argument as String?) {
            null, "yaml" -> return Result.success(organizationManager.toYaml())
            "json" -> return Result.success(organizationManager.toJson())
            "csv" -> return Result.success(organizationManager.toCSV())
            else -> return Result.failure(InvalidOutputFormatException())
        }
    }
}
